boll-branch/nuxt-segment
# Support | HEAD-2177 | Consolidate segment packages

Ticket URL: https://bollandbranch.atlassian.net/browse/HEAD-2177

Testing:
- Checkout this branch locally
- Use `yarn link` (`npm link` has never worked too well for me without causing headaches) within the root directory of this locally checked out branch - You should see the following messages if all worked correctly:

```bash
success Registered "@boll-and-branch/nuxt-segment".
info You can now run `yarn link "@boll-and-branch/nuxt-segment"` in the projects where you want to use this package and it will be used instead.
```

- Open up `boll-branch-nacelle` repository and run `yarn link @boll-branch/nuxt-segment` 
- Once this is done, check the `node_modules/@boll-branch` directory to make sure you see this icon next to the nuxt-segment package:
<img width="166" alt="Screen Shot 2022-11-18 at 4 50 57 PM" src="https://user-images.githubusercontent.com/24883328/202825591-ebdcee6e-bd42-4277-ba99-436356efcfa6.png">

- Add the following key to the `.env` file within `boll-branch-nacelle`

```
# Ping in slack if you don't have/can't find this value
SEGMENT_TRACKING_CDN="<segment_cdn>"
```

Replace the CDN value within the `static/bb-segment.js` file as well, search for `.src` within this file to find where this is.

***CDN and the write key MUST have matching sources**

- Run `npm run dev` from the nacelle project and check the network logs for segment requests:
	- Segment event requests should be firing
	- Segment request payloads should contain the correct key
